cmd/syncthing/support_bundle.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"archive/zip"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/syncthing/syncthing/lib/config"
+)
+
+// AddBufferFile adds new json to fileByte(name, bytes.Buffer)
+func addBufferFile(arrayFile []fileByte, name string, json []byte) []fileByte {
+	var singleFile fileByte
+	singleFile.name = name
+	singleFile.buf.Write(json)
+	arrayFile = append(arrayFile, singleFile)
+	return arrayFile
+}
+
+// GetRedactedConfig redacting some parts of config
+func getRedactedConfig(s *apiService) config.Configuration {
+	rawConf := s.cfg.RawCopy()
+	rawConf.GUI.APIKey = "REDACTED"
+	rawConf.GUI.Password = "REDACTED"
```

Comment:
Only do this if the User/Password value are set, to preserve the fact that no authentication is set when it isn't.

```
guiCfg := s.cfg.GUI()
if guiCfg.Password != "" {
   guiCfg.Password = "REDACTED"
}
...
```